controllers/slicegateway/reconciler.go

```diff
@@ -192,8 +192,75 @@ func (r *SliceGwReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 			)
 			return ctrl.Result{}, err
 		}
+		//reconcile the available nodes
+		if err := r.reconcileNodes(ctx, sliceGw); err != nil {
+			return ctrl.Result{}, err
+		}
 	}
 
+	isClient := sliceGw.Status.Config.SliceGatewayHostType == "Client"
+	if isClient {
```

Comment:
This whole block can be a separate func. r.ReconcileGwHeadlessSvc()

Reply:
yeah that would make code clean